tests/ci/cdk/cdk/ssm/general_test_run_ssm_document.yaml

```diff
@@ -14,31 +14,55 @@ mainSteps:
         # Codebuild just "stops" the instance calling the script, so "trap cleanup" is not executed.
         - shutdown -P +60
         - sudo -i
+        - systemctl stop apt-daily.timer
         - export DEBIAN_FRONTEND=noninteractive
         - export CPU_TYPE=$(dpkg --print-architecture)
+        - export SOURCE={SOURCE}
         # if we have a cpu type of x86, we want linux-x86
         - if [ "${CPU_TYPE}" = amd64 ]; then export CPU_ARCH=linux-x86; export AWS_CLI_PREFIX=x86_; sudo sh -c "echo 1 > /sys/devices/system/cpu/intel_pstate/no_turbo"; fi
         # if we have a cpu type of arm, we want linux-aarch
         - if [ "${CPU_TYPE}" = arm64 ]; then export CPU_ARCH=linux-aarch; export AWS_CLI_PREFIX=aarch; fi
         # install aws-cli
+        - killall apt apt-get
+        - apt-get update
         - apt-get -y install unzip
         - curl "https://awscli.amazonaws.com/awscli-exe-linux-${AWS_CLI_PREFIX}64.zip" -o "awscliv2.zip"
         - unzip awscliv2.zip
         - ./aws/install
-        - git clone {GITHUB_REPO} aws-lc-pr
-        - cd aws-lc-pr
-        - git checkout {COMMIT_ID}
+        # Check if the source code is on S3, otherwise treat the source as a PR.
+        - >
+          if [ "$(expr substr "$SOURCE" 1 16)" = "aws-lc-codebuild" ]; then
+            aws s3api get-object --bucket {S3_BUCKET} --key "${SOURCE##{S3_BUCKET}/}" aws-lc-pr.zip
+            unzip aws-lc-pr.zip
+          else
+            git clone {SOURCE} aws-lc-pr
+            cd aws-lc-pr
+            git fetch origin pull/{PR_NUM}/head:temp
+            git checkout temp
```

Comment:
For sanity can this log the output of git show, and then also check that the current commit matches the expected source.

Reply:
Sure, will do